### PR TITLE
add gflags

### PR DESCRIPTION
**Pull request by mo271**

### CODE REVIEW

**Comment by mo271** [comment]
The failing CiFuzz should be fixed by adding the corresponding dependency in https://github.com/google/oss-fuzz/blob/master/projects/libjxl/Dockerfile. Will do as soon as this is merged

**Comment by mo271** [comment]
I proposed using this instead of abseil-cpp, so to close #999.

File: third_party/CMakeLists.txt [hunk 1]
```diff
@@ -153,6 +153,28 @@ if (JPEGXL_ENABLE_VIEWERS OR NOT JPEGXL_ENABLE_SKCMS)
...
+  add_subdirectory(gflags)
+  configure_file("${CMAKE_CURRENT_SOURCE_DIR}/gflags/COPYING.txt"
+                 ${PROJECT_BINARY_DIR}/LICENSE.gflags COPYONLY)
+else()
+  # Use sytem libgflags-dev
+  find_package(gflags)
+  if (NOT gflags_FOUND)
+    message(FATAL_ERROR
+      "Gflags not found, install libgflags-dev or download gflags source to"
+      "third_partygflags from https://github.com/gflags/gflags. You can use"
```
Comment (veluca93) [inline third_party/CMakeLists.txt:169]: Typo

**Review by veluca93 — APPROVED** [review]

File: third_party/CMakeLists.txt (hunk 1 above)
Comment (mo271) [reply third_party/CMakeLists.txt]: fixed

**Review by veluca93 — APPROVED** [review]

**Comment by mo271** [comment]
working on fixing the CIFuzz check: https://github.com/google/oss-fuzz/pull/7063

File: .github/workflows/release.yaml [hunk 2]
```diff
@@ -35,6 +35,7 @@ jobs:
           doxygen \
           libbrotli-dev \
           libgdk-pixbuf2.0-dev \
+          libgflag-dev \
```
Comment (eustas) [inline .github/workflows/release.yaml:38]: Should it be "libgflags-dev"?

Comment (mo271) [reply .github/workflows/release.yaml:38]: yes.
Thanks for fixing it in https://github.com/libjxl/libjxl/commit/a46cbcc65ca1f22820c25343330a441cf9f658e3 !

**Pull request merged**